leaking-vars-tests

ArmorDarks opened the pull request:
Added requested in #9 files

I hope I didn't miss anything.

Should I update `compare.js`, or create new one in `tests/leaking-vars`?

vitkarpov commented on Gruntfile.js:
```diff
@@ -21,7 +21,10 @@ module.exports = function(grunt) {
         files: {
-          'tests/_output.html' : ['tests/input.html']
+          'tests/_output.html' : ['tests/input.html'],
+          'tests/_output.html' : ['tests/input.html'],
```
seems it's the same as above

vitkarpov commented:
I think we could use the same `compare.js`, at least for a while

vitkarpov commented:
It would be great to see test in `compare.js` which fails, so I could merge the pr and try to explore the problem. The solution is when the test starts to pass.

ArmorDarks commented:
Ops, fixed doubled line :)

I've added `compare.js`, but I'm not sure about it's syntax, since I'm not very familiar with expect. Please, check it out.

vitkarpov commented on tests/leaking-vars/compare.js:
```diff
@@ -0,0 +1,16 @@
+var sample1 = fs.readFileSync('tests/leaking-vars/output1.html').toString();
+var sample2 = fs.readFileSync('tests/leaking-vars/output1.html').toString();
+var test1 = fs.readFileSync('tests/leaking-vars/_output2.html').toString();
+var test2 = fs.readFileSync('tests/leaking-vars/_output2.html').toString();
```
we need to compare `output1` width `_output1` and `output2` with `_output2`.

```
var sample1 = fs.readFileSync('tests/leaking-vars/output1.html').toString();
var sample2 = fs.readFileSync('tests/leaking-vars/output2.html').toString();
var test1 = fs.readFileSync('tests/leaking-vars/_output1.html').toString();
var test2 = fs.readFileSync('tests/leaking-vars/_output2.html').toString();
```

?

vitkarpov commented:
Seems ok :+1: Let's fix the comparing as I mentioned above and let's merge this.

ArmorDarks commented:
Uh, I'd want to say that I'm just still learnin js, but it seems to be more related to my inattentiveness :(

I hope that time I didn't miss anything.
